### PR TITLE
Fix/18885 migrate avatar token

### LINKED ISSUE #18885: Migrate components to TS: AvatarToken
### Description
Migrate `AvatarToken` component to TypeScript

### Technical Details
- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` follow [TS conventions](https://www.notion.so/Use-TypeScript-9ec7bd8fc8cf4d82b81cd8943c4707cd?pvs=4#7e6410b21100455a9a08442108f19d78)
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation 

### Acceptance Criteria
- Component has been converted to TypeScript

### PR DESCRIPTION
**Pull request by thebinij**
## Explanation
* Closes #18885 

### Before

https://github.com/MetaMask/metamask-extension/assets/93417120/351d02e4-6b55-418a-9505-5fc5699ac026

### After

https://github.com/MetaMask/metamask-extension/assets/93417120/4a884625-9d50-4eda-85c7-79c2b82578a8

## Manual Testing Steps
No functional changes

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/avatar-token/avatar-token.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    const handleOnError = () => {
+      setShowFallback(true);
+    };
+
+    const fallbackString = name?.[0] ?? '?';
+
+    return (
+      <AvatarBase
+        ref={ref}
+        size={size as unknown as AvatarBaseSize}
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-token/avatar-token.tsx:44]: Change back to size and add `AvatarTokenSize` as a union type to `AvatarBase`
```suggestion
        size={size}
```

**Review by georgewrmarshall — COMMENTED** [review]
Hey @thebinij, I've checked out your branch and tried a few things to get the types to work but I can't seem to find a reliable pattern. I think for now we just make `AvatarBaseProps` accept all variant size types as a union type for now and update it in the future once we find a scalable pattern to use.

File: ui/components/component-library/avatar-token/avatar-token.tsx (hunk 1 above)
Comment (thebinij) [reply ui/components/component-library/avatar-token/avatar-token.tsx:44]: Agreed, let's do that.

File: ui/components/component-library/avatar-token/avatar-token.test.tsx [hunk 2]
```diff
@@ -19,42 +19,66 @@ describe('AvatarToken', () => {
 
   it('should render correctly', () => {
     const { getByTestId, container } = render(
-      <AvatarToken {...args} data-testid="avatar-token" />,
+      <AvatarToken children={undefined} {...args} data-testid="avatar-token" />,
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-token/avatar-token.test.tsx:22]: Can we remove children for the types?

**Comment by georgewrmarshall** [comment]
Hey @thebinij, Thanks for your time and effort in creating this pull request. As you're probably aware we have identified an issue related to the typing of the `Box` component, specifically with the polymorphic `as` prop. To address this issue, we have an open ticket at https://github.com/MetaMask/metamask-extension/issues/19239 and a draft pull request at https://github.com/MetaMask/metamask-extension/pull/19363 that needs to be merged before we can proceed with your PR. Thanks for your patience

File: ui/components/component-library/avatar-network/avatar-network.types.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ export enum AvatarNetworkSize {
  * Props for the AvatarNetwork component
  */
 export interface AvatarNetworkStyleUtilityProps
-  extends Omit<AvatarBaseStyleUtilityProps, 'size'> {
+  extends Omit<AvatarBaseStyleUtilityProps, 'size' | 'children'> {
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/avatar-network/avatar-network.types.ts:16]: I think this makes sense with my limited TypeScript knowledged brain. We are adjusting the types at the same level we are increasing the specificity of the component.

**Review by georgewrmarshall — APPROVED** [review]
Looks alll G to ME! Thanks for your contribution @thebinij and for supporting @garrettbear 💯

**Comment by georgewrmarshall** [comment]
> `ButtonLink` lint error that is probably coming from the ButtonBase update and hopefully will be fixed when converting ButtonLink to TS.
> 
> Screenshot 2023-07-25 at 3 59 55 PM

Looks like we are safe for the most part as the only use in tsx files are in stories which are ignored from the linter. I think we can prioritize buttons next

Screenshot 2023-07-25 at 5 29 05 PM

**Review by garrettbear — APPROVED** [review]

**Pull request merged**